torch/distributed/_tensor/placement_types.py

```diff
@@ -60,9 +60,17 @@ def _split_tensor(
             self.dim <= tensor.ndim
         ), f"Sharding dim {self.dim} greater than tensor ndim {tensor.ndim}"
 
-        # chunk tensor over dimension `dim` into n slices with padding if necessary
+        # chunk tensor over dimension `dim` into n slices
         tensor_list = list(torch.chunk(tensor, num_chunks, dim=self.dim))
-        # compute the chunk size inline with ``torch.chunk``
+
+        # if no need to have padding or tensor dim size is evenly sharded already
+        # we can return early.
+        if not with_padding or tensor.size(self.dim) % num_chunks == 0:
+            if contiguous:
+                tensor_list = [t.contiguous() for t in tensor_list]
+            return tensor_list, []
```

Comment:
directly returning seems problematic according to the test:
```
l = torch.chunk(torch.rand(5, 2), 4)  # yields 3 smaller tensors of size (2, 2), (2, 2), (1, 2)
l[3]  # index out of range
```

Reply:
In this case, with_padding would be True and tensor.size(self.dim) % num_chunks won't be 0, right? So it shouldn't go in this if, if I am understanding this correctly.

Reply:
The failed test case has `with_padding=False`.

Reply:
fixed the tests